test/test_cuda.py

```diff
@@ -4148,8 +4153,23 @@ def test_no_triton_on_import(self):
         self.assertEqual(rc, "False", "Triton was imported when importing torch!")
 
 
+class myTest(TestCase):
+    @optims([optim for optim in optim_db if optim.optim_cls in
+             [torch.optim.Adam, torch.optim.AdamW, torch.optim.SGD]], dtypes=[torch.float32])
+    def test_grad_scaling_autocast_fused_optimizers(self, device, dtype, optim_info):
+        device = torch.device(device)
+        optim_cls = optim_info.optim_cls
+        optim_inputs = _get_optim_inputs_including_global_cliquey_kwargs(device, dtype, optim_info, skip=('differentiable'))
+        for optim_input in optim_inputs:
+            for optimizer_params, optimizer_kwargs, separate_unscale in list(product(optim_input.params,
+                                                                                     optim_input.kwargs, (False, True))):
+                optimizer = optim_cls(optimizer_params, optimizer_kwargs)
+                with self.subTest(optim=optimizer, kwargs=optimizer_kwargs, separate_unscale=separate_unscale):
+                    self._grad_scaling_autocast_fused_optimizers(self, optimizer, optimizer_kwargs=optimizer_kwargs,
```

Comment:
We should no longer need the _grad_scaling_autocast_fused_optimizers helper method anymore, as it can be refactored into one combined function.

I’ll review more in detail after a rebase and this comment is addressed!

Reply:
Thanks for the feedback! I rebased but I'm not sure how to implement the refactoring completely. I understand that we want to essentially copy the logic from `_grad_scaling_autocast_fused_optimizers` into the new test, I just wish I could read up on the optimizer usage. Is there some documentation that I can refer to? Or is the source code the best place to look.

Reply:
Feel free to use https://github.com/pytorch/pytorch/commit/94af62b000a9256591d91e17ad7002b0d99a05cc as an example!